### DISCUSSION TITLE
[service-contracts] `ServiceProviderInterface` doesn't expose exceptions

### DISCUSSION THREAD

**Discussion by etshy** [symfony/symfony#57648] [Q&A]
Upgrading to symfony 6 from 5.4, the interface `ContainerInterface` are not exposed anymore, so I planned to change the injection to `ServiceProviderInterface`.
And I noticed the interface `ServiceProviderInterface` doesn't expose the exception that can get thrown on the `get()` method, when a requested service is not present for example.

Is that intended ?

ps: not really a bug per so so posting it here.

**Comment by MatTheCat** [comment]
`ServiceProviderInterface` extends [PSR-11](https://www.php-fig.org/psr/psr-11/)’s `ContainerInterface` which documents both `NotFoundExceptionInterface` and `ContainerExceptionInterface`.

As [the `throws` tag is inherited](https://docs.phpdoc.org/guide/guides/inheritance.html#methods), `ServiceProviderInterface` also inherits these.

**Reply by etshy** [reply]
Maybe It's a PhpStorm issue then ? 
But as the docblock is re-defined in the `ServiceProviderInterface` without `@inheritdoc` and with no `@throws`, Phpstorm doesn't warn about them.
Thanks for the answer, I wasn't aware docblock should "auto-inherit" some information

**Reply by MatTheCat** [reply]
My PhpStorm does tell me when I don’t catch these exceptions :thinking: 

I’m using the 2024.1 version; maybe yours is not up-to-date?

**Reply by etshy** [reply]
I use 2024.1.1 and I have this weak warning on a method that use the `get()`
![image](https://github.com/symfony/symfony/assets/5531204/869016de-ba29-45c9-9c88-e2299b63fa83)

**Reply by MatTheCat** [reply]
I suppose this is right: it is not **your** function which will throw. However, you should have a warning on the `get` call saying you didn’t handle the exceptions it can throw.

**Reply by etshy** [reply]
Yeah I guess you can see it that way, but effectively, if my method doesn't catch them, they will be "thrown" from my method too.
But I guess it's more of a PHPStorm issue anyway, referring to the docblock docs you linked earlier.

**Reply by MatTheCat** [reply]
Seems like this can be configured with the “Call tree analysis depth” setting in PhpStorm: https://blog.jetbrains.com/phpstorm/2018/04/configurable-unchecked-exceptions/

**Reply by etshy** [reply]
Didn't know about that, thanks.
unfortunately, it doesn't help, I tried to set it up to 3 (max level) and in method the throws were still not detected.
And the Exception are not extended from the "unchecked Exceptions"
